packages/fast-components-react-msft/src/flyout/flyout.tsx

```diff
@@ -0,0 +1,201 @@
+                        FlyoutAxisPositioningMode[this.props.verticalPositioningMode]
+                    ]
+                }
+                verticalThreshold={this.props.verticalThreshold}
+                viewport={this.props.viewport}
+                style={{
+                    height: this.props.height,
+                    width: this.props.width,
+                }}
+            >
+                {this.props.children}
+            </ViewportPositioner>
+        );
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidMount(): void {
+        if (canUseDOM() && this.props.onDismiss) {
+            window.addEventListener("keydown", this.handleWindowKeyDown);
+            window.addEventListener("click", this.handleWindowClick);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(prevProps: Partial<FlyoutProps>): void {
+        if (canUseDOM()) {
+            if (!prevProps.onDismiss && this.props.onDismiss) {
+                window.addEventListener("keydown", this.handleWindowKeyDown);
+                window.addEventListener("click", this.handleWindowClick);
+            } else if (prevProps.onDismiss && !this.props.onDismiss) {
+                window.removeEventListener("keydown", this.handleWindowKeyDown);
+                window.removeEventListener("click", this.handleWindowClick);
+            }
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentWillUnmount(): void {
+        if (canUseDOM() && this.props.onDismiss) {
+            window.removeEventListener("keydown", this.handleWindowKeyDown);
+            window.removeEventListener("click", this.handleWindowClick);
+        }
+    }
+
+    private generateManagedClassNames(): ViewportPositionerClassNameContract {
+        return {
+            viewportPositioner: get(this.props.managedClasses, "flyout", ""),
+            viewportPositioner__left: get(this.props.managedClasses, "flyout__left", ""),
+            viewportPositioner__right: get(
+                this.props.managedClasses,
+                "flyout__right",
+                ""
+            ),
+            viewportPositioner__top: get(this.props.managedClasses, "flyout__top", ""),
+            viewportPositioner__bottom: get(
+                this.props.managedClasses,
+                "flyout__bottom",
+                ""
+            ),
+            viewportPositioner__horizontalInset: get(
+                this.props.managedClasses,
+                "flyout__horizontalInset",
+                ""
+            ),
+            viewportPositioner__verticalInset: get(
+                this.props.managedClasses,
+                "flyout__verticalInset",
+                ""
+            ),
+        };
+    }
+
+    private handleWindowClick = (event: MouseEvent): void => {
+        if (typeof this.props.onDismiss === "function" && !this.props.disabled) {
+            this.props.onDismiss(event);
```

Comment:
Whats the story if the user clicks *inside* the flyout? Do we still want to dismiss the flyout?

Reply:
touché - good catch.

Reply:
^ just to be clear no, correct? The user should be able to interact with content inside the flyout - that's the point of a dialog-type flyout

Reply:
correct.